### PR TITLE
feat: create a reuseable dialog component

### LINKED ISSUE #49: [FEAT] - Components → DialogBox
## Description
Create a Dialog Box component that includes a header with text and description, and a footer that accepts multiple buttons.

## Acceptance Criteria
- Implement a Dialog Box component with a header that includes `headerText` and `description`.
- The footer should accept multiple buttons.
- Ensure the Dialog Box component is visually appealing and responsive.
- Maintain accessibility standards with appropriate aria attributes.

## Requirements
The Dialog Box component should support the following props:
- `headerText: string` - Required text to be displayed in the header.
- `description?: string` - Optional description to be displayed under the header text.
- `buttons: React.ReactNode[]` - An array of button components to be displayed in the footer.

## Testing
- Verify that the Dialog Box component renders correctly with the provided headerText, description, and buttons.
- Test that the footer can accommodate multiple buttons.
- Ensure responsiveness and visual appeal across different screen sizes.
- Conduct accessibility testing to ensure screen reader compatibility.

## Tasks
- [ ] **Component Development**: Develop the Dialog Box component based on the specified requirements.
- [ ] **Props Implementation**: Implement support for `headerText`, `description`, and `buttons` props.
- [ ] **Styling**: Ensure consistent styling for the header, description, and footer.
- [ ] **Unit Testing**: Write unit tests to validate component behavior.
- [ ] **Accessibility Testing**: Verify accessibility features are properly implemented.
- [ ] **Documentation**: Update documentation with usage examples and prop details.

#### Dialog Card
Dialog card

## Link
[Figma Link](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2225-7657&t=XOIp4pqGzAyZNcxj-4)

### PR DESCRIPTION
**Pull request by Adelosoye-dev**
## Description

Add a Dialog Box component that includes a header with text and description, and a footer that accepts multiple buttons.

Closes #49 

## Changes Proposed

Create a Dialog Box component that includes a header with text and description, and a footer that accepts multiple buttons.

### Requirements

The Dialog Box component should support the following props:

- headerText: string - Required text to be displayed in the header.

- description?: string - Optional description to be displayed under the header text.

- buttons: React.ReactNode[] - An array of button components to be displayed in the footer.

### What were you told to do?

[FEAT] - Component → Dialogbox

Create a Dialog Box component that includes a header with text and description, and a footer that accepts multiple buttons.

### What did you do?

Created a reusable `Dialog` component. 

The `Dialogbox` component displays a card with a header that includes headerText, description and footer that accepts multiple buttons.. It is fully responsive and designed to meet accessibility standards. It receives the following props:

#### Props:

- **headerText**

  - Type: `string`
  - Required: `true`
  - Description:  The heater text for the dialog component.
  
- **description**

  - Type: `string`
  - Required: `false`
  - Description: Optional description to be displayed under the header text.
   
  
- **buttons**

  - Type: `array`
  - Required: `true`
  - Description: An array of button components to be displayed in the footer.
  
### Usage

```tsx
import Dialog from "../components/Dialog/Dialog";

<Dialog
              headerText="Delete Organization"
              description="Are you sure you want to delete Jolly organization ? All of your data will be permanently 
              removed. This action cannot be undone."
              footer={[
                <button
                  key="cancel"
                  className="bg-red-500 text-white py-2 px-4 rounded"
                  onClick={handleCloseDialog}
                >
                  Cancel
                </button>,
                <button
                  key="Delete"
                  className="bg-blue-500 text-white py-2 px-4 rounded"
                >
                  Delete
                </button>,
              ]}
            />
```

-  Dialog Component Screenshot

Screenshot 2024-07-20 233433

- Figma link

[https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=2225-7657&t=XOIp4pqGzAyZNcxj-4](url)

### CODE REVIEW

**Comment by chumex412** [comment]
You have conflicts

**Comment by chumex412** [comment]
Resolve the conflicts

File: app/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  description?: string;
+  footer: React.ReactNode[];
+};
+const Dialog = ({ headerText, description, footer }: DialogProperties) => {
+  return (
+    <div className="flex flex-col p-6 gap-4 bg-white rounded-[6px] max-w-[512px] w-auto">
+      <header className="flex flex-col gap-2">
+        <h2 className="text-lg font-semibold text-primary">{headerText}</h2>
+        {description && (
+          <p className="font-normal text-sm leading-[1.05rem] text-muted-foreground text-pretty">
```
Comment (chumex412) [inline app/components/Dialog/Dialog.tsx:14]: Why are you creating your own dialog? Weren't you told to use ShadCN dialog component?

**Review by chumex412 — CHANGES_REQUESTED** [review]

File: app/components/Dialog/Dialog.tsx (hunk 1 above)
Comment (Adelosoye-dev) [reply app/components/Dialog/Dialog.tsx:14]: In the issue requirements, I wasn't asked to use Shadcn-UI.

The Shadcn-UI Dialog component requires a close icon to close the dialog and a trigger button to display it, but these were not mentioned in the requirements and also not in the figma file.

I have attached the figma link for the component in the PR description for your reference.

Should I still go ahead and implement them anyway?

Comment (chumex412) [reply app/components/Dialog/Dialog.tsx]: Meet @SirhmVFX on slack to install it. he's @fullstackmechanic on slack

**Review by shalomtaiwo — CHANGES_REQUESTED** [review]
Still didn't fix what was requested. Use Shadcn

**Comment by Adelosoye-dev** [comment]
i have made the changes

**Review by mrcoded — APPROVED** [review]

**Pull request merged**